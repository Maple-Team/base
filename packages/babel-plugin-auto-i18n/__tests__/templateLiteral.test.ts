import { hash } from '@liutsing/node-utils'
import { getTransformCode } from './test-help'

describe('arrow function declaration scenarios - templateLiteral cases', () => {
  it('handle templateLiteral case 1', () => {
    const sourceCode = `export const Component = () => {
        const plateNo = 'No.123';
        const var1 = \`车牌号: \${plateNo}\`;
        return <div>{var1}</div>
      }
      `
    const result = getTransformCode(sourceCode, `${hash(sourceCode)}.tsx`)
    expect(result?.code).toMatchSnapshot()
  })

  it('handle templateLiteral case 2', () => {
    const sourceCode = `export const Component = () => {
        const plateNo = 'No.123';
        return <div>{\`车牌号: \${plateNo}\`}</div>
      }
      `
    const result = getTransformCode(sourceCode, `${hash(sourceCode)}.tsx`)
    expect(result?.code).toMatchSnapshot()
  })
  it('handle templateLiteral case 3', () => {
    const sourceCode = `export const Component = () => {
        const plateNo = 'No.123';
        const price = '￥20W';
        const var1 = \`车牌号: \${plateNo}, \${price}\`;
        return <div>{var1}</div>
      }
      `
    const result = getTransformCode(sourceCode, `${hash(sourceCode)}.tsx`)
    expect(result?.code).toMatchSnapshot()
  })

  it('handle templateLiteral case 4', () => {
    const sourceCode = `export const Component = () => {
        const plateNo = 'No.123';
        const price = '￥20W' + 'abc';
        const var1 = \`车牌号: \${plateNo}, \${price}\`;
        return <div>{var1}</div>
      }
      `
    const result = getTransformCode(sourceCode, `${hash(sourceCode)}.tsx`)
    expect(result?.code).toMatchSnapshot()
  })
})
