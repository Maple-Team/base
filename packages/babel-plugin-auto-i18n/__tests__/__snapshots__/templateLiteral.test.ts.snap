// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrow function declaration scenarios - templateLiteral cases handle templateLiteral case 1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var plateNo = 'No.123';
  var var1 = t('key_车牌号: ', {
    key1: plateNo
  });
  return <div data-i18n=\\"[\\\\\\"key_\\\\u8F66\\\\u724C\\\\u53F7: \\\\\\"]\\">{var1}</div>;
};"
`;

exports[`arrow function declaration scenarios - templateLiteral cases handle templateLiteral case 2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var plateNo = 'No.123';
  return <div data-i18n=\\"[\\\\\\"key_\\\\u8F66\\\\u724C\\\\u53F7: \\\\\\"]\\">{t('key_车牌号: ', {
      key1: plateNo
    })}</div>;
};"
`;

exports[`arrow function declaration scenarios - templateLiteral cases handle templateLiteral case 3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var plateNo = 'No.123';
  var price = '￥20W';
  var var1 = t('key_车牌号: , ', {
    key1: plateNo,
    key2: price
  });
  return <div data-i18n=\\"[\\\\\\"key_\\\\u8F66\\\\u724C\\\\u53F7: , \\\\\\"]\\">{var1}</div>;
};"
`;

exports[`arrow function declaration scenarios - templateLiteral cases handle templateLiteral case 4 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var plateNo = 'No.123';
  var price = '￥20W' + 'abc';
  var var1 = t('key_车牌号: , ', {
    key1: plateNo,
    key2: price
  });
  return <div data-i18n=\\"[\\\\\\"key_\\\\u8F66\\\\u724C\\\\u53F7: , \\\\\\"]\\">{var1}</div>;
};"
`;

exports[`arrow function declaration scenarios - templateLiteral cases handle templateLiteral case 5 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var plateNo = 'No.123';
  return <div data-i18n=\\"[\\\\\\"key_\\\\u5DF2\\\\u6025\\\\u505C\\\\\\"]\\">{t('key_已急停', {
      key1: plateNo
    })}</div>;
};"
`;
